### ISSUE TITLE
(More) functional Python with the `itertools` and `functools` libraries

### ISSUE THREAD

**Issue by bertrand-caron** [BrisPy/talks#51]
Hi all, 

As discussed with @cjrh and @rstuart85 at the last MeetUp, I'd love to give a talk on "(More) functional Python with the `itertools` and `functools` libraries" if you think that would be of interest to the audience.

The outline would most likely look something like that:

1. What is functional programming? Why bother? When is it appropriate (or "don't try to put a functional peg in an object-oriented hole")?
2. Why Python can support functional programming: functions are first-level citizens, support for unnamed functions (lambdas), etc.
3. Walkthrough of functools (reduce, groupby, memoization, etc.) with real-life examples
4. Walkthrough of itertools (or "when life gives you lemons, make combinatorial lemonade") with real-life examples.
If extra-time:
5. Functional features still missing in Python (for instance, pattern matching/guards)

January will be a bit of a full-on month for me (both from a personal and professional standpoint), so I'd rather not present in February, but if you guys are really in a pickle I can try to make it work.

Best, Bertrand C.

**Comment by kaiw** [comment]
Definitely of interest! I've marked this as needs scheduling, so once you get time we can pick a month.

**Comment by bertrand-caron** [comment]
@kaiw I'd love to present this in April, would that work?

**Comment by kaiw** [comment]
@bertrand-caron Sorry, I've been delaying only because we apparently got a couple of volunteers for April at the meetup last Wednesday, and I wasn't sure whether or not you were one of those.

If you talked to Kieran on Wednesday about this then yes! absolutely! If not then I'll have to get back to you.

**Comment by KieranTHE** [comment]
Yes, Bertrand was one of the speakers that said he was keen. Definitely book it in.

**Comment by kaiw** [comment]
Excellent, thanks! @bertrand-caron I've booked you in for the general slot on April 8 (https://github.com/BrisPy/talks/blob/master/README.md).

**Comment by bertrand-caron** [comment]
@kaiw  and @KieranTHE : Sorry for the slow answer. I look forward to presenting on April the 8th :) 
When do you need a talk abstract by?

**Comment by kaiw** [comment]
@bertrand-caron We don't need an abstract at all! We really just need your talk title for the meetup announcement.

**Comment by kaiw** [comment]
@bertrand-caron Given the current environment, we're going to cancel BrisPy meetups until further notice. I really hope that you'll still be keen to present when we reschedule after the current meeting restrictions, etc. have lifted.

**Comment by bertrand-caron** [comment]
@kaiw I figured that much :) I would love to present this when things have calmed down, or even attempt presenting it online (zoom, etc.) at some point if that is an avenue you are considering.

**Comment by kaiw** [comment]
We've briefly discussed online presentation, but I don't think any of the organisers have the spare time to figure out the logistics of that right now. If that changes in the future (particularly if we end up in a very prolonged period of meeting restrictions) then I'll let you know. Thanks!
